File: learn-scala-sample/src/main/scala/com/jwd/scala/sample/functional/Fold.scala
package com.jwd.scala.sample.functional

object Fold {
	val numbers = List(5, 15, 30, 20, 35, 45)
	
	def main(args : Array[String]) {
	  println(numbers.foldLeft(0) {(l, r) => (l + r)})
	}
}

File: learn-scala-sample/src/main/scala/com/jwd/java/sample/fold/Fold.java
package com.jwd.java.sample.fold;

public class Fold {
	public static void main(String[] args) {
		int[] numbers = {5, 15, 30, 20, 35, 45};
		int total = 0;
		
		for (int i = 0; i < numbers.length; i++) {
			total += numbers[i];
		}
		
		System.out.println(total);
	}
}
